engine/runtime-parser/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -156,1892 +155,31 @@ object IR {
-    ) extends Expression
-        with IRKind.Primitive {
-      override protected var id: Identifier = randomId
-
-      /** Creates a copy of `this`.
-        *
-        * @param name the name being bound to
-        * @param expression the expression being bound to `name`
-        * @param location the source location that the node corresponds to
-        * @param passData the pass metadata associated with this node
-        * @param diagnostics compiler diagnostics for this node
-        * @param id the identifier for the new node
-        * @return a copy of `this`, updated with the specified values
-        */
-      def copy(
-        name: IR.Name                        = name,
-        expression: Expression               = expression,
-        location: Option[IdentifiedLocation] = location,
-        passData: MetadataStorage            = passData,
-        diagnostics: DiagnosticStorage       = diagnostics,
-        id: Identifier                       = id
-      ): Binding = {
-        val res = Binding(name, expression, location, passData, diagnostics)
-        res.id = id
-        res
-      }
-
-      /** @inheritdoc */
-      override def duplicate(
-        keepLocations: Boolean   = true,
-        keepMetadata: Boolean    = true,
-        keepDiagnostics: Boolean = true,
-        keepIdentifiers: Boolean = false
-      ): Binding =
-        copy(
-          name = name.duplicate(
-            keepLocations,
-            keepMetadata,
-            keepDiagnostics,
-            keepIdentifiers
-          ),
-          expression = expression.duplicate(
-            keepLocations,
-            keepMetadata,
-            keepDiagnostics,
-            keepIdentifiers
-          ),
-          location = if (keepLocations) location else None,
-          passData =
-            if (keepMetadata) passData.duplicate else MetadataStorage(),
-          diagnostics =
-            if (keepDiagnostics) diagnostics.copy else DiagnosticStorage(),
-          id = if (keepIdentifiers) id else randomId
-        )
-
-      /** @inheritdoc */
-      override def setLocation(location: Option[IdentifiedLocation]): Binding =
-        copy(location = location)
-
-      /** @inheritdoc */
-      override def mapExpressions(fn: Expression => Expression): Binding = {
-        copy(name = name.mapExpressions(fn), expression = fn(expression))
-      }
-
-      /** @inheritdoc */
-      override def toString: String =
-        s"""
-        |IR.Expression.Binding(
-        |name = $name,
-        |expression = $expression,
-        |location = $location
-        |passData = ${this.showPassData},
-        |diagnostics = $diagnostics,
-        |id = $id
-        |)
-        |""".toSingleLine
-
-      /** @inheritdoc */
-      override def children: List[IR] = List(name, expression)
 
-      /** @inheritdoc */
-      override def showCode(indent: Int): String =
-        s"${name.showCode(indent)} = ${expression.showCode(indent)}"
-    }
-  }
```

Comment:
Two thousand lines less. That's is an achievement of its own!

Reply:
It's not really less. Just that IR.scala will no longer be such a beast.

Reply:
Sure, still it means way less scrolling.